lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -36,37 +36,40 @@
   state:
     description:
       - Create or delete Lambda function
-    required: false
     default: present
     choices: [ 'present', 'absent' ]
   runtime:
     description:
       - The runtime environment for the Lambda function you are uploading. Required when creating a function. Use parameters as described in boto3 docs.
         Current example runtime environments are nodejs, nodejs4.3, java8 or python2.7
-    required: true
+      - Required when C(state=present)
```

Comment:
This is meant to make the "Required when" part of the description (whllst the required boolean defaults to false).  Does required: when ... work?  If so then we should do a fix to a bunch of modules which are using the same system.